### PR TITLE
OS#16050168: fix jit crash for splitscope using 'with' statements

### PR DESCRIPTION
**Pull request by sigatrev**
When 'with' statements are used in inside param scopes when param and body scopes are split, there will not always be a local closure generated. See attached test for an example case.

The IRBuilder would attempt to build a RegOpnd for the closure, and would assert on a nullptr sym. The backward pass would later AV on that nullptr sym.

This commit instead assigns nullptr to the param closure when there is no local closure, which is the behavior of InterpreterStackFrame::OP_BeginBodyScope()

### CODE REVIEW

**Comment by sigatrev** [comment]
@akroshg The local closure is not generated because of the logic in ByteCodeGenerator::ProcessScopeWithCapturedSym. The scope containing the captured sym is the 'with' scope, which is marked as MustInstantiate and thus can't merge.

File: lib/Backend/IRBuilder.cpp [hunk 1]
```diff
@@ -6815,23 +6815,34 @@ IRBuilder::BuildEmpty(Js::OpCode newOpcode, uint32 offset)
...
         this->SetParamScopeDone();
 
+        IR::Opnd * localClosureOpnd;
+        if (this->m_func->GetLocalClosureSym() != nullptr)
+        {
+            localClosureOpnd = IR::RegOpnd::New(this->m_func->GetLocalClosureSym(), TyVar, this->m_func);
+        }
+        else
+        {
+            localClosureOpnd = IR::IntConstOpnd::New(0, TyVar, this->m_func);
```
Comment (aneeshdk) [inline lib/Backend/IRBuilder.cpp]: Can we assert here that either scoepslotarraysize is 0 or there is scope object?

Comment (aneeshdk) [inline lib/Backend/IRBuilder.cpp]: Can you assert or fail fast here that scoprslotarraysize is 0 and no scope object?

**Comment by aneeshdk** [comment]
LGTM

**Pull request merged**